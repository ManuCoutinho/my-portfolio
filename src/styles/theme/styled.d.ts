import 'styled-components'

declare module 'styled-components' {
  export interface DefaultTheme extends CustomTheme {
    title: string
    colors: {
      black: string
      white: string
      background: string
      body: string
      primary: string
      secondary: string
      text: string
      title: string
      info: string
      highlight: string
      carousel_1: string
      carousel_2: string
      carousel_3: string
      carousel_4: string
      bg_card: string
      glass: string
    }
    font: {
      text: string
      title: string
    }
    fontSize: {
      xsm: string
      sm: string
      md: string
      lg: string
      xl: string
      xlg: string
    }
    fontWeight: {
      light: number
      normal: number
      semibold: number
      bold: number
    }
    transition: string
    radius: {
      small: string
      default: string
      medium: string
      full: string
    }
  }

  export interface CustomTheme {
    title: string
    colors: {
      black: string
      white: string
      background: string
      body: string
      primary: string
      secondary: string
      text: string
      title: string
      info: string
      highlight: string
      carousel_1: string
      carousel_2: string
      carousel_3: string
      carousel_4: string
      bg_card: string
      glass: string
    }
  }
}
