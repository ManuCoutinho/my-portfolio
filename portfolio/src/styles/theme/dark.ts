export const dark = {
  title: 'light',
  colors: {
    black: '#090B09',
    white: '#FFFCF2',
    background: '#1C221C',
    body: '#121713',
    primary: '#1A7444',
    secondary: '#12BF62',
    text: '#e8e4df',
    title: '#434a3c',
    info: '#AEAFAC',
    highlight: '#5AFF15',
  },
  font: {
    text: "'Montserrat Alternates', sans-serif",
    title: "'Josefin Sans', sans-serif",
  },
  fontSize: {
    xsm: '0.75em',
    sm: '0.875em',
    md: '1em',
    lg: '1.5em',
    xl: '2.5em',
    xlg: '3.5em',
  },
  fontWeight: {
    light: 100,
    normal: 400,
    semibold: 600,
    bold: 700,
  },
  transition: 'all 0.2s ease-in-out',
  radius: {
    default: '16px',
    medium: '25px',
    full: '50%',
  },
}
