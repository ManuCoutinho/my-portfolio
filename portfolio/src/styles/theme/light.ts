export const light = {
  title: 'light',
  colors: {
    black: '#090B09',
    white: '#FAFAFF',
    background: '#E38F77',
    body: '#F7F6F3',
    primary: '#817b92',
    secondary: '#6e758d',
    text: '#242b31',
    title: '#0A0A0A',
    info: '#56667c',
    highlight: '#26F7F0',
  },
  font: {
    text: "'Montserrat Alternate', sans-serif",
    title: "'Josefin Sans', sans-serif",
  },
  fontSize: {
    xsm: '0.75em',
    sm: '0.875em',
    md: '1em',
    lg: '1.5em',
    xl: '2.5em',
    xlg: '3.5em',
  },
  fontWeight: {
    light: 100,
    normal: 400,
    semibold: 600,
    bold: 700,
  },
  transition: 'all 0.2s',
  radius: '8px',
}
