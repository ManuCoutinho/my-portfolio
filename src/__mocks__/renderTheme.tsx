import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
