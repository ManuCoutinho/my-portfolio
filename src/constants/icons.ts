export default {
	github: 'iconoir:github-outline',
	linkedin: 'ph:linkedin-logo',
	twitter: 'radix-icons:twitter-logo',
	email: 'ep:message',
	sun: 'ph:sun-dim',
	moon: 'ph:moon-stars',
	close: 'ph:x-circle',
	x: 'ph:x',
	hamburger: 'ph:list-bold',
	chevron_left: 'ph:caret-left',
	chevron_right: 'ph:caret-right',
	link_external: 'codicon:link-external',
	ts: 'file-icons:typescript-alt',
	styled: 'file-icons:styledcomponents',
	next: 'file-icons:nextjs',
	vite: 'file-icons:vite',
	jamstack: 'simple-icons:jamstack',
	tailwind: 'file-icons:tailwind',
	mui: 'simple-icons:mui',
	chakra: 'simple-icons:chakraui',
	jest: 'file-icons:jest',
	bs: 'fa-brands:bootstrap',
	css: 'akar-icons:css-fill',
	git: 'fa-brands:git-alt',
	gh: 'fa-brands:github',
	html: 'akar-icons:html-fill',
	jquery: 'mdi:jquery',
	react: 'fa-brands:react',
	sass: 'fa-brands:sass',
	js: 'ion:logo-javascript',
	node: 'mdi:nodejs',
	sb: 'file-icons:storybook',
	sentry: 'file-icons:sentry',
	graphql: 'mdi:graphql',
	figma: 'ph:figma-logo',
	testing: 'simple-icons:testinglibrary',
	docs: 'tabler:brand-storybook',
	docker: 'devicon-plain:docker-wordmark',
	devto: 'ph:dev-to-logo',
	actions: 'devicon-plain:githubactions',
	angular: 'mdi:angularjs',
	material: 'devicon-plain:angularmaterial'
}
