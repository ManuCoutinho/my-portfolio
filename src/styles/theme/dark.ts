import { lighten } from 'polished'

const dark = {
  title: 'dark',
  colors: {
    black: '#090B09',
    white: '#FFFCF2',
    background: '#090B09',
    body: '#090B09',
    primary: '#1A7444',
    secondary: '#12BF62',
    text: '#DEE2E6',
    title: '#02C39A',
    info: '#78a590',
    highlight: 'rgba(15, 255, 103, 0.2)',
    active: '#37d27f',
    carousel_1: '#2D392D',
    carousel_2: '#080a08',
    carousel_3: 'rgba(10, 13, 10, 0.1)',
    carousel_4: 'rgba(255, 255, 255, 0.02)',
    bg_card: '#1F1F1F',
    glass: 'rgba(9, 11, 9, 0.65)',
    bg_modal: '#292929',
    disabled: lighten(0.5, '#1c1c1c'),
    border_1: 'rgba(15, 255, 103, 1)',
    border_2: '#37d27f'
  }
} as const

export default dark
