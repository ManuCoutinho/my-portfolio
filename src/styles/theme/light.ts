import { lighten } from 'polished'
const light = {
	title: 'light',
	colors: {
		black: '#353131',
		white: '#f3faf0',
		background: '#e48c59',
		body: '#f3faf0',
		primary: 'rgb(201, 100, 81)',
		secondary: '#ae6a40',
		text: '#003844',
		title: '#353131',
		info: '#8A5536',
		highlight: 'rgb(58, 12, 163, 0.5)',
		carousel_1: '#DEDDDB',
		carousel_2: '#f5f5f5',
		carousel_3: '#CBCAC7',
		carousel_4: '#f5f5f5',
		active: '#F05D51',
		bg_card: '#f8e3dd',
		glass: 'rgba(243, 250, 240, 0.1)',
		bg_modal: '#e19b76',
		disabled: lighten(0.5, '#56667c')
	}
} as const

export default light
