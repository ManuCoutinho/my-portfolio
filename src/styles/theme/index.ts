import { DefaultTheme, CustomTheme } from 'styled-components'
import dark from './dark'
import light from './light'

export type Theme = typeof defaultTheme

const defaultTheme = {
	font: {
		cursive:
			"'Montserrat Alternates', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif",
		regular:
			"'Josefin Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif"
	},
	fontSize: {
		xsm: '0.5rem', //8px
		sm: '0.875rem', //14px
		md: '1rem', //16px
		wd: '1.25rem', //20px
		lg: '1.5rem', //24px
		xl: '2rem', //36px
		xlg: '3rem' //48px
	},
	fontWeight: {
		thin: 100,
		light: 200,
		regular: 400,
		semibold: 600,
		bold: 700
	},
	radius: {
		sm: '8px',
		md: '16px',
		pill: '25px',
		full: '50%'
	}
} as const

function combineTheme(theme: CustomTheme): DefaultTheme {
	return { ...defaultTheme, ...theme }
}
export { combineTheme, dark, light }
