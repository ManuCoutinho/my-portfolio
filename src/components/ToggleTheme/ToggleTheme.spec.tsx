import { screen, fireEvent } from '@testing-library/react'
import { render } from '../../styles/theme/renderTheme'
import { ToggleTheme } from '.'
import icons from 'constants/icons'

describe('<ToggleTheme/>', () => {
  it('should render a  Toggle Theme component correctly', () => {
    render(<ToggleTheme />)
    expect(screen.getByRole('switch')).toBeInTheDocument()
  })
  it('should render a  Toggle Theme with right Icon', () => {
    render(<ToggleTheme />)
    expect(screen.getByRole('switch').firstChild).not.toHaveAttribute(
      'icon',
      icons.sun
    )
  })
  it('should render a  switch icon correctly', () => {
    render(<ToggleTheme />)
    const button = screen.getByRole('switch')
    fireEvent.click(button)
    expect(screen.getByRole('switch').firstChild).toHaveAttribute(
      'icon',
      icons.sun
    )
  })
  it('should the Toggle Theme match to snapshot', () => {
    const { container } = render(<ToggleTheme />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
