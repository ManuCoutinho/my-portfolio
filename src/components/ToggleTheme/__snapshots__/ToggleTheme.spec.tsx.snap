// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleTheme/> should the Toggle Theme match to snapshot 1`] = `
<button
  aria-label="switch theme"
  class="sc-58177537-0 DDfXa"
  role="switch"
  type="button"
>
  <iconify-icon
    icon="ph:sun-dim"
  />
</button>
`;
